nd4j/nd4j-backends/nd4j-backend-impls/nd4j-cuda/src/main/java/org/nd4j/linalg/jcublas/buffer/BaseCudaDataBuffer.java

```diff
@@ -671,25 +673,24 @@ public void copyDataFromSrc(Pointer pointer,long length, long srcOffset,long dst
      */
     public void set(int[] data, long length, long srcOffset, long dstOffset) {
         // TODO: make sure getPointer returns proper pointer
-
+        lazyAllocateHostPointer();
+        allocator.synchronizeHostData(this);
         switch (dataType()) {
             case BOOL: {
                 val pointer = new BytePointer(ArrayUtil.toBytes(data));
                 copyDataFromSrc(pointer,length,offset,dstOffset);
             }
             break;
+
+            case UBYTE: {
+                data = ArrayUtil.cutBelowZero(data);
+            }
```

Comment:
Is this explicitly meant to fall through to the next case? If so, put a note here.